src/main/php/xp/frontend/Resolver.class.php

```diff
@@ -0,0 +1,35 @@
+<?php namespace xp\frontend;
+
+use lang\IllegalArgumentException;
+use text\json\{Json, StreamInput};
+
+class Resolver {
+  private $fetch, $registry;
+
+  public function __construct(Fetch $fetch, string $registry= 'https://registry.npmjs.org') {
+    $this->fetch= $fetch;
+    $this->registry= rtrim($registry, '/').'/';
+  }
+
+  public function version(string $library, string $constraint): string {
+    $info= Json::read(new StreamInput($this->fetch->get($this->registry.$library)));
+
+    // TBI: Create constraints
```

Comment:
This needs a well-tested implemenation for:

* [ ] Selecting the latest version - `library`
* [ ] Selecting a specific version - `library@1.11.2`
* [ ] Selecting the next signification release - `library@~1.11`
* [ ] Selecting the next release in the same series - `library@^1.11`
* [ ] Selecting any patch level release in 1.11 series - `library@1.11.*`

See https://getcomposer.org/doc/articles/versions.md